mclagsyncd/mclaglink.cpp

```diff
@@ -87,163 +112,258 @@ void MclagLink::getVidByBvid(std::string &bvid, std::string &vlanid)
+            {
+                if (0 == temp.find("Ethernet"))
+                {
+                    continue;
+                }
+                if (isolate_dst_port.length())
+                {
+                    isolate_dst_port = isolate_dst_port + ',' + temp;
+                }
+                else
+                {
+                    isolate_dst_port = temp;
+                }
+            }
 
-        p_acl_table_tbl->set(acl_name, acl_attrs);
+            fvts.emplace_back("DESCRIPTION", "Isolation group for MCLAG");
+            fvts.emplace_back("TYPE", "bridge-port");
+            fvts.emplace_back("PORTS", isolate_src_port);
+            fvts.emplace_back("MEMBERS", isolate_dst_port);
 
-        acl_table_is_added = 1;
-        /*End create ACL table*/
+            p_iso_grp_tbl->set("MCLAG_ISO_GRP", fvts);
+            SWSS_LOG_NOTICE("Isolation group created with ports %s and members %s",
+                    isolate_src_port.c_str(),
+                    isolate_dst_port.c_str());
+        }
     }
+    else
+    {
+        mclag_sub_option_hdr_t *op_hdr = NULL;
+        string isolate_src_port;
+        string isolate_dst_port;
+        char * cur = NULL;
+        string acl_name = "mclag";
+        string acl_rule_name = "mclag:mclag";
+        vector<FieldValueTuple> acl_attrs;
+        vector<FieldValueTuple> acl_rule_attrs;
+        std::string acl_key = std::string("") + APP_ACL_TABLE_TABLE_NAME + ":" + acl_name;
+        std::string acl_rule_key = std::string("") + APP_ACL_RULE_TABLE_NAME + ":" + acl_rule_name;
+        static int acl_table_is_added = 0;
+
+        cur = msg;
+
+        /*get isolate src port infor*/
+        op_hdr = reinterpret_cast<mclag_sub_option_hdr_t *>(static_cast<void *>(cur));
+        cur = cur + MCLAG_SUB_OPTION_HDR_LEN;
+        isolate_src_port.insert(0, (const char*)cur, op_hdr->op_len);
+
+        cur = cur + op_hdr->op_len;
+
+        /*get isolate dst ports infor*/
+        op_hdr = reinterpret_cast<mclag_sub_option_hdr_t *>(static_cast<void *>(cur));
+        cur = cur + MCLAG_SUB_OPTION_HDR_LEN;
+        isolate_dst_port.insert(0, (const char*)cur, op_hdr->op_len);
+
+        if (op_hdr->op_len == 0)
+        {
+            /* If dst port is NULL, delete the acl table 'mclag' */
+            p_acl_table_tbl->del(acl_name);
+            acl_table_is_added = 0;
+            SWSS_LOG_NOTICE("set port isolate, src port: %s, dst port is NULL",
+                    isolate_src_port.c_str());
+            return;
+        }
+
+        SWSS_LOG_NOTICE("set port isolate, src port: %s, dst port: %s",
+                isolate_src_port.c_str(), isolate_dst_port.c_str());
+
+        if (acl_table_is_added == 0)
+        {
+            /*First create ACL table*/
+            FieldValueTuple desc_attr("policy_desc", "Mclag egress port isolate acl");
+            acl_attrs.push_back(desc_attr);
+
+            FieldValueTuple type_attr("type", "L3");
```

Comment:
@gitsabari From the [HLD](https://github.com/Azure/SONiC/blob/master/doc/mclag/Sonic-mclag-hld.md#726-l2-bum-flooding-and-port-isolation) it looks that ACL is needed for L2 traffic isolation, but table type is L3.
Why do we use `L3`?
Is it a bug?